### DISCUSSION TITLE
anyone get this working with waterfox?

### DISCUSSION THREAD

**Discussion by squid-slime** [filips123/PWAsForFirefox#280] [Q&A]
ive burned my search engine out looking, ive also looked through the code (not a coder but never hurt to do so), i did manage to install along side firefox allowing me to add links from firefox and opening the webapp in a waterfox container but so far having difficulty connecting the browser extension to the firefox-pwa application

using waterfox-g G5.1.1 (64-bit) on arch kde plasma

**Comment by filips123** [comment]
Do you want to use Waterfox as a main browser or as an app browser that runs your PWAs?


If the former, this is probably because Waterfox uses different paths for native messaging manifests. You can simply link the path where Waterfox looks for manifests to the directory where manifests are normally stored for Firefox:

```bash
ln -s ~/.mozilla/native-messaging-hosts ~/.waterfox/native-messaging-hosts
sudo ln -s /usr/lib/mozilla/native-messaging-hosts /usr/lib/waterfox/native-messaging-hosts
```

Alternatively, you can copy `/usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json` to `/usr/lib/waterfox/native-messaging-hosts/firefoxpwa.json`.


If you want to use Waterfox as an app browser, you need to download [Waterfox tarball](https://www.waterfox.net/download/), extract it and copy it to `$XDG_DATA_HOME/firefoxpwa/runtime` (if `$XDG_DATA_HOME` is set) or `$HOME/.local/share/firefoxpwa/runtime` (if it is not), and symlink `waterfox` binary as `firefox` inside that directory.

**Reply by squid-slime** [reply]
had this working last week with symbolic links but after an update to firefox-pwa and waterfox it seems the browser no longer opens to the predetermined sites, isteand opening a blank 4by4 or what ever ever size id made the window prior, im giving the tarball method ago although i am wondering how easy would it be to go without the add-on/extension and have a system gui or even a basic cli to add address key words etc, can i have some suggestions on where to start as in the directories thatll be useful?

**Reply by filips123** [reply]
Does it work if you try to use normal Firefox instead of Waterfox? Also, instead of using the extension, you can use [the console program](https://github.com/filips123/PWAsForFirefox/tree/main/native#usage) for things like installing a new web app.

**Comment by SeleneCosmia** [comment]
Tried this and found that /usr/lib/waterfox doesn't exist and I couldn't find anything similar anywhere on my system

**Reply by filips123** [reply]
How did you install Waterfox?

**Reply by SeleneCosmia** [reply]
Through my package manager (pikaur/pacman), which is probably why, because the only available package that was up to date was the pre compiled binary version of waterfox. I also have firefox so I'm using the extension with firefox instead for simplicity's sake.

**Reply by filips123** [reply]
Maybe it's installed in `/opt/waterfox`? Otherwise, I don't know where else would it be... You can also check if your `waterfox` binary is a symlink to anywhere.
